mantle/kola/tests/misc/network.go

```diff
@@ -299,79 +320,134 @@ var (
-                  ;;
-                *)
-                  ;;
-               esac
-            done
-            echo -e "default dev: $default_device ($profile_name)\nsecondary dev: $secondary_device ($secondary_profile_name)"
-            
-            mac=$(sudo nmcli -g GENERAL.HWADDR dev show $default_device | sed -e 's/\\//g')
-          
-            # delete old bridge if it exists
-            ovs-vsctl --if-exists del-br brcnv
-            
-            # make bridge
-            nmcli conn add type ovs-bridge conn.interface brcnv
-            nmcli conn add type ovs-port conn.interface brcnv-port master brcnv
-            nmcli conn add type ovs-interface \
-                           conn.id brcnv-iface \
-                           conn.interface brcnv master brcnv-port \
-                           ipv4.method auto \
-                           ipv4.dhcp-client-id "mac" \
-                           connection.autoconnect no \
-                           802-3-ethernet.cloned-mac-address $mac
-  
-            # make bond
-            nmcli conn add type ovs-port conn.interface bond0 master brcnv ovs-port.bond-mode balance-slb
-            nmcli conn add type ethernet conn.interface $default_device master bond0
-            nmcli conn add type ethernet conn.interface $secondary_device master bond0
-            nmcli conn down "$profile_name" || true
-            nmcli conn mod "$profile_name" connection.autoconnect no || true
-            nmcli conn down "$secondary_profile_name" || true
-            nmcli conn mod "$secondary_profile_name" connection.autoconnect no || true
-            if ! nmcli conn up brcnv-iface; then
-                nmcli conn up "$profile_name" || true
-                nmcli conn mod "$profile_name" connection.autoconnect yes
-                nmcli conn up "$secondary_profile_name" || true
-                nmcli conn mod "$secondary_profile_name" connection.autoconnect yes
-                nmcli c delete $(nmcli c show |grep ovs-cnv |awk '{print $1}') || true
-            else
-                nmcli conn mod brcnv-iface connection.autoconnect yes
-            fi
-          else
-              echo "ovs bridge already present"
-          fi
+		set -ex
+
+		connections_folder=$(NetworkManager --print-config | sed -nr "/^\[keyfile\]/ { :l /^path[ ]*=/ { s/.*=[ ]*//; p; q;}; n; b l;}")
+		if [[ ! -d "$connections_folder" ]]; then
+		  connections_folder=/etc/NetworkManager/system-connections
+		fi
+		rm -rf ${connections_folder}/*
+
+		if [[ ! -f /boot/mac_addresses ]] ; then
+		  echo "no mac address configuration file found .. exiting"
+		  exit 0
+		fi
+
+		if [[ -d /var/ovsbond ]]; then
+		  echo "Loading OVS old profile"
+		  cp -r /var/ovsbond/* ${connections_folder}
+		  systemctl restart NetworkManager
+		fi
+
+		if [[ $(nmcli conn | grep -c ovs) -eq 0 ]]; then
+		  echo "configure ovs bonding"
+		  ovs-vsctl --if-exists del-br brcnv
```

Comment:
So...personally long term I think we should kick ovs back out of RHCOS and run from a privileged container again.

If we do that, then this test would need to change.

As is right now, PRs to this repo only test FCOS (xref https://github.com/openshift/release/pull/20929 ) so this would need to be tested manually.  I assume you've done so?

Reply:
>So...personally long term I think we should kick ovs back out of RHCOS and run from a privileged container again.

I agree, We have plans to do so in the future.

>As is right now, PRs to this repo only test FCOS (xref openshift/release#20929 ) so this would need to be tested manually. I assume you've done so?

yes